test/model.jl

```diff
@@ -492,6 +515,85 @@ function test_copy_direct_mode()
     @test_throws ErrorException JuMP.copy(model)
 end
 
+function test_copy_filter()
+    copy_model_style_mode(true, MOIU.AUTOMATIC, true)
+end
+    
+function test_copy_filter_array()
+    model = Model()
+    model.optimize_hook = dummy_optimizer_hook
+    data = DummyExtensionData(model)
+    model.ext[:dummy] = data
+    @variable(model, x[i=1:2] ≥ 0, container=Array)
+    @constraint(model, cref[i=1:2], x[i] == 1, container=Array)
+    filter_constraints = (cr) -> cr != cref[1]
+    new_model, reference_map = JuMP.copy_model(model, filter_constraints=filter_constraints)
```

Comment:
Test that the number of AffineExpression-in-EqualTo constraints in the model is one. I think this test will fail because of my comment on `copy_model`.